feat(icons): Add sort remove icon

AllanJard opened the pull request:
## What is the purpose of this pull request?
- [X] New Icon
- [ ] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [ ] Other:

### Description

A new icon for use alongside `arrow-down-wide-narrow` and `arrow-up-narrow-wide`.

I'm not 100% on the name that I've chosen for the icon...!

### Icon use case

I'm the author of [DataTables](https://datatables.net) and I'm using Lucide icons for a new extension that shows options in the header of columns. Lucide has icons for sort-asc and sort-desc, but not one for removing the sorting from a column (which is functionality I want to have as an option). I created this icon to fill the gap and thought I'd share it back to see if you wanted to have it in this amazing icon set you've created.

This is a screenshot of my use case:

![image](https://github.com/user-attachments/assets/1958dce7-0d21-45fa-80dc-6cff197a18da)

Second use case could see it being used to indicate no sorting on a column, rather than a removal action.

### Alternative icon designs

I haven't come up with any alternatives at the moment.

## Icon Design Checklist 

### Concept

- [X] I have provided valid use cases for each icon.
- [X] I have not added any a brand or logo icon.
- [X] I have not used any hate symbols.
- [X] I have not included any religious or political imagery.

### Author, credits & license

- [X] The icons are solely my own creation.
- [ ] The icons were originally created in #<issueNumber> by @<githubUser>
- [X] I've based them on the following Lucide icons:  `arrow-down-wide-narrow` and `arrow-up-narrow-wide`.
- [ ] I've based them on the following design: <!-- provide source URL and license permitting use -->

### Naming 

- [X] I've read and followed the [naming conventions](https://lucide.dev/guide/design/icon-design-guide#naming-conventions)
- [X] I've named icons by what they are rather than their use case.
- [X] I've provided meta JSON files in `icons/[iconName].json`.

### Design

- [X] I've read and followed the [icon design guidelines](https://lucide.dev/guide/design/icon-design-guide)
- [X] I've made sure that the icons look sharp on low DPI displays.
- [X] I've made sure that the icons look consistent with the icon set in size, optical volume and density.
- [X] I've made sure that the icons are visually centered.
- [X] I've correctly optimized all icons to three points of precision.

## Before Submitting 

- [X] I've read the [Contribution Guidelines](https://github.com/lucide-icons/lucide/blob/main/CONTRIBUTING.md).
- [X] I've checked if there was an existing PR that solves the same issue.

AllanJard committed:
feat(icons): Add sort remove icon

AllanJard committed:
Renaming icon to be more inkeeping with lucide naming requirements

jguddas commented on icons/bars-x.svg:
```diff
@@ -0,0 +1,17 @@
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <path d="m 3,20 8,-8" />
+  <path d="M 11,20 3,12" />
+  <path d="M 11,4 H 21" />
+  <path d="M 11,8 H 21" />
+  <path d="m 15,12 h 6" />
```
Optimized paths.
```suggestion
  <path d="m11 20-8-8" />
  <path d="M11 4h10" />
  <path d="M11 8h10" />
  <path d="M15 12h6" />
  <path d="m3 20 8-8" />
```

AllanJard committed:
Incorporating feedback from @jguddas

Optimise paths
Include authors of arrow-down-wide-narrow and arrow-up-narrow-wide

AllanJard commented:
Many thanks for the feedback! I've added them in.

ericfennis commented:
Why not using `list-x`? 
https://lucide.dev/icons/list-x

AllanJard commented:
I felt that the X in the bottom left fit with the other icons a bit more, but yep, totally understand if you close this in preference for `list-x`.

ericfennis committed:
Merge branch 'main' into main

ericfennis commented:
What is your opinion about this @karsa-mistmere ?

AllanJard commented:
I made a few more I needed as well:

![image](https://github.com/user-attachments/assets/812a8ce7-99c4-4461-97db-aecce1570655)

The top two exist, this PR adds the one I've got for "Clear sort" there. I've added the icons for "Remove from sort" and the two "Add Sort..." icons. I can add them to the PR if it is of interest. But equally if not, no worries. I'm just delighted that Lucide exists :-)

ericfennis commented:
`list-x` is too similar. Fills the same purpose. Closing this one for now.
